rust/agents/validator/src/submit.rs

```diff
@@ -37,7 +37,22 @@ impl ValidatorSubmitter {
         let span = info_span!("ValidatorSubmitter");
         let reorg_period = Some(self.reorg_period);
         tokio::spawn(async move {
-            let mut current_index = self.checkpoint_syncer.latest_index().await?.unwrap_or_default();
+            // Ensure that the outbox has > 0 messages before we enter the main
+            // validator submit loop. This is to avoid an underflow / reverted
+            // call when we invoke the `outbox.latest_checkpoint()` method,
+            // which returns the **index** of the last element in the tree
+            // rather than just the size.  See
+            // https://github.com/abacus-network/abacus-monorepo/issues/575 for
+            // more details.
+            let mut outbox_count = self.outbox.count().await?;
```

Comment:
@mattiecnvr Not sure if there's a more elegant loop construction in Rust than what I ended up with. What I wanted to write was something along the lines of:

```c++
for (int c = 0; c <= 0; c = self.outbox.count().await?) {
  info!(c);
  tokio::time::sleep(Duration::from_secs(self.interval)).await;
}
```

Reply:
Yeah, no, I thought there was at first glance but this pattern is something that is inherently confusing when written in the C-syntax so there is intentionally no direct analog.

It probably would be a little more canonical to do
```rs
loop {
  let outbox_count = self.outbox.count().await?;
  if outbox_count > 0 { break }
  info!(outbox_count, "waiting for non-zero outbox size");
  sleep(Duration::from_secs(self.interval)).await;
}
```

Why is this _slightly_ better:
1) it avoids `mut` which is a rust sin, but not end of the world
2) it avoids writing the same fetch statement twice
3) `loop` + `break` is pretty well accepted for weird loop conditions